### PR TITLE
Implement task and subteam

### LINKED ISSUE #7: Update due date failed
Skærmbillede 2022-10-17 kl  20 20 02

To replicate:
Run https://github.com/rlindsberg/2207ID-SEP-Business-Management-System/commit/858a8ada8da940d46ba3e1e8d88b7e8a7cd02704

Run the following in postman:
1. create request
2. create new event
3. create new subteam
4. create new task
5. update due date

### PR DESCRIPTION
**Pull request by HaidaLu**
closes #7 
1. Implement the Task class
2. Implement the Subteam class
3. Implement add budget comment

### CODE REVIEW

File: backend/src/main/java/com/SEP/BusinessManagementSystem/Controller/TaskController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+public class TaskController {
+    private final TaskService taskService;
+
+    @Autowired
+    public TaskController(TaskService taskService) {
+        this.taskService = taskService;
+    }
+
+
+    @GetMapping("/get/{Id}")
```
Comment (rlindsberg) [inline backend/src/main/java/com/SEP/BusinessManagementSystem/Controller/TaskController.java]: Can you change Id to id?

File: backend/src/main/java/com/SEP/BusinessManagementSystem/Controller/TaskController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public ResponseEntity<?> getTaskById(@PathVariable Long Id){
+        return ResponseEntity.ok().body(taskService.getTaskById(Id));
+    }
+
+    @PostMapping("/create/{Id}")
```
Comment (rlindsberg) [inline backend/src/main/java/com/SEP/BusinessManagementSystem/Controller/TaskController.java]: here as well?

File: backend/src/main/java/com/SEP/BusinessManagementSystem/Entity/Task.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package com.SEP.BusinessManagementSystem.Entity;
+
+import javax.persistence.*;
+import java.time.LocalDate;
+
+@Entity
+@Table
+public class Task {
+    @Id
+    private Long id;
```
Comment (rlindsberg) [inline backend/src/main/java/com/SEP/BusinessManagementSystem/Entity/Task.java:13]: This id should be generated, right?

**Review by rlindsberg — COMMENTED** [review]
Just some questions...

File: backend/src/main/java/com/SEP/BusinessManagementSystem/Controller/TaskController.java (hunk 2 above)
Comment (HaidaLu) [reply backend/src/main/java/com/SEP/BusinessManagementSystem/Controller/TaskController.java]: fixed!

File: backend/src/main/java/com/SEP/BusinessManagementSystem/Entity/Task.java (hunk 3 above)
Comment (HaidaLu) [reply backend/src/main/java/com/SEP/BusinessManagementSystem/Entity/Task.java:13]: can be auto-generated now!

**Comment by rlindsberg** [comment]
The tests did not pass, please see https://github.com/rlindsberg/2207ID-SEP-Business-Management-System/issues/7

**Review by rlindsberg — APPROVED** [review]
Good job! You can merge now!

**Pull request merged**